sim_pipeline/sim_lsst_galaxy_galaxy_lens_simulation.py

```diff
@@ -0,0 +1,283 @@
+        n = len(pipeline_cut)
+        index = random.randint(0, n - 1)
+        source = pipeline_cut[index]
+        return source
+
+    def get_params(self):
+        """
+        Method to get the galaxy parameters to be used in the simulation
+        return the band information g,r,i; (including the magnitude, size, shape, and position under random scatter)
+               the distance between the galaxy and observer;
+               the redshift of the galaxy;
+               the stellar mass; (for velocity dispersion)
+
+        """
+        source= self.get_source()
+        stellar_mass = source['stellar_mass']
+        redz = source['z']
+        distan = [self.cosmo.angular_diameter_distance(redz).value]    # distance between galaxy and observer
+        size_arcsec = source['angular_size'] / constants.arcsec   # convert radian to arc seconds
+        mag_g, mag_r, mag_i = source['mag_g'], source['mag_r'], source['mag_i']
+        mag_apparent = source['M']
+        e = source['ellipticity']
+        phi = np.random.uniform(0, np.pi)
+        e1 = e * np.cos(phi)
+        e2 = e * np.sin(phi)
+        if self.galaxy_type == 'blue':
+            n_sersic = 1
+        else:
+            n_sersic = 4
+        center_x, center_y = np.random.uniform(-self.position_scatter, self.position_scatter), np.random.uniform(
+            -self.position_scatter, self.position_scatter)
+        kwargs_galaxy_g = [{'magnitude': mag_g, 'R_sersic': size_arcsec, 'n_sersic': n_sersic,
+                            'e1': e1, 'e2': e2, 'center_x': center_x, 'center_y': center_y}]
+        kwargs_galaxy_r = [{'magnitude': mag_r, 'R_sersic': size_arcsec, 'n_sersic': n_sersic,
+                            'e1': e1, 'e2': e2, 'center_x': center_x, 'center_y': center_y}]
+        kwargs_galaxy_i = [{'magnitude': mag_i, 'R_sersic': size_arcsec, 'n_sersic': n_sersic,
+                            'e1': e1, 'e2': e2, 'center_x': center_x, 'center_y': center_y}]
+        return [kwargs_galaxy_g, kwargs_galaxy_r, kwargs_galaxy_i], distan, source['z'], stellar_mass
+
+def get_vd_from_stallermass(smass):
+    """
+    function for calculate the velocity from the staller mass for velocity dispersion
+
+    :param smass: stellar mass in the unit of solar mass,
+
+    return vdmass: the velocity dispersion ("km/s")
+    2.32,0.24 is the parameters from [1] table 2
+    [1]:Auger, M. W., et al. "The Sloan Lens ACS Survey. X. Stellar, dynamical, and total mass correlations of massive early-type galaxies." The Astrophysical Journal 724.1 (2010): 511.
+    """
+    stellarmass = smass
+    vdmass = (np.power(10,2.32) * np.power(stellarmass/1e11, 0.24))
+    return vdmass
+
+configblue = LenstronomyConfig(pipeline=skycon, mag_cut=22, z_min=0.8, z_max=2, galaxy_type='blue', position_scatter=0.5)
+configred = LenstronomyConfig(pipeline=skycon, mag_cut=20, z_min=0.1, z_max=0.4, galaxy_type='red', position_scatter=0.0)
+
+def process_image(ax, sim_g, sim_r, sim_i, cosmolo): #
```

Comment:
this function mixes a lot of different things. I.e. source, lens light, mass model, lenstronomy configurations and plotting. I would suggest having a class or function that each deals with only one aspect, i.e. Source(), Lens(), then another class that casts the complete information into lenstronomy. And another class that does the plotting. This gives the best way to extend the code without breaking it.